src/main/kotlin/org/rust/ide/lineMarkers/CargoTestRunLineMarkerContributor.kt

```diff
@@ -17,22 +17,39 @@ import org.rust.cargo.runconfig.command.CargoExecutableRunConfigurationProducer
 import org.rust.cargo.runconfig.test.CargoBenchRunConfigurationProducer
 import org.rust.cargo.runconfig.test.CargoTestLocator
 import org.rust.cargo.runconfig.test.CargoTestRunConfigurationProducer
+import org.rust.cargo.runconfig.test.getDoctestCtx
 import org.rust.lang.core.psi.RsElementTypes.IDENTIFIER
 import org.rust.lang.core.psi.RsFunction
 import org.rust.lang.core.psi.ext.RsNameIdentifierOwner
 import org.rust.lang.core.psi.ext.RsQualifiedNamedElement
 import org.rust.lang.core.psi.ext.elementType
+import org.rust.lang.doc.psi.RsDocCodeFence
+import org.rust.lang.doc.psi.RsDocCodeFenceStartEnd
+import org.rust.lang.doc.psi.RsDocElementTypes.DOC_DATA
 import javax.swing.Icon
 
 class CargoTestRunLineMarkerContributor : RunLineMarkerContributor() {
     override fun getInfo(element: PsiElement): Info? {
-        if (element.elementType != IDENTIFIER) return null
-        val parent = element.parent
-        if (parent !is RsNameIdentifierOwner || element != parent.nameIdentifier) return null
-        if (parent is RsFunction && CargoExecutableRunConfigurationProducer.isMainFunction(parent)) return null
+        val target = when (element.elementType) {
+            DOC_DATA -> {
+                val parent = element.parent as? RsDocCodeFenceStartEnd ?: return null
+                val fence = parent.parent as? RsDocCodeFence ?: return null
+                if (fence.start != parent) return null
+                // Ignore invalid fences
+                if (fence.end == null) return null
```

Comment:
Btw non-closed code fences are valid from the cargo/rustdoc PoV

Reply:
Good to know. I only added this condition because in the IDE it seemed to me that it was creating the marker too soon, but since it's valid even if there's no end triple backtick, this should be fine.